src/averageAges.js

```diff
@@ -37,7 +59,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenArray = people.filter(({ sex }) => sex === SEX_FEMALE);
+
+  const filteredWomenByChildren = withChildren
+    ? womenArray.filter(({ name }) =>
+      people.some(({ mother }) => mother === name))
+    : womenArray;
+
+  return getAverageAge(filteredWomenByChildren);
```

Comment:
The logic for filtering women by children is good, but the check could be moved inside the main filter method to avoid creating an additional array if 'withChildren' is false.